test/PowerShellEditorServices.Test/Debugging/DebugServiceTests.cs

```diff
@@ -264,8 +264,8 @@ public async Task DebuggerSetsAndClearsFunctionBreakpoints()
                 }).ConfigureAwait(true);
 
             Assert.Equal(2, breakpoints.Length);
-            Assert.Equal("Write-Host", breakpoints[0].Name);
-            Assert.Equal("Get-Date", breakpoints[1].Name);
+            Assert.Equal("Write-Host", breakpoints.ElementAt(0).Name);
```

Comment:
Same double enumeration warning here, this line and the line below will invoke the producing code twice first the first element

(I know this is just tests and doesn't necessarily impact anything, just worth pointing out because the behavior is not exactly intuitive)

Reply:
What's the right way to do it?

Reply:
I would say `ToArray` it first is the most common answer. You could `foreach` it and save the first and second instance, but that gets a lot more complicated the more you want to do with it. Like further on when you do `Count()` that'll enumerate everything to get the count, then when you do `ElementAt(0).LineNumber` it'll run `Create` again, and again when you do `ElementAt(0).Verified`, etc